### ISSUE TITLE
SpaceAroundOperator should allow newlines as well

### ISSUE THREAD

**Issue by srawlins** [brigade/scss-lint#598]
Right now SpaceAroundOperator has only two `style` config choices:
`one_space` or `no_space`. How about a `one_space_or_newline` like SpaceAfterPropertyColon has?

This would support the uses of string concatenation in order to keep a maximum width on files, like:

```scss
p:before {
  content: 'Something crazy long and weird and longer than '
      + '80 characters.';
}
```

I think the allowed whitespace of `one_space_or_newline` would allow:

* one space character on each side of an operator, or
* one space character on one side of the operator and a newline followed by any number of space characters on the other side (indentation is not measured).

```scss
a {
  content: 'hello ' + 'world';
  font-family: 'hello ' +
      'world';
  something-else: 'hello '
      + 'world';
}
```

(all legal SCSS.)

So, not allowed:

* an operator on a line by itself
* multiple newlines
* tab characters

What do you think? Happy to do the work.

**Comment by sds** [comment]
In the case of operators, I would consider a newline to be allowed by the `one_space`/`no_space` styles, since there are situations where both sides of the operator cannot fit on a single line.

Perhaps the answer here is to allow newlines for these styles.

**Comment by srawlins** [comment]
I'm into it. Do you think the procedure should then be: if there is a newline on _either_ side of an operator (anywhere amongst whitespace characters), then we don't lint that operator any further, and let anything through?

Or be a little more strict?

**Comment by sds** [comment]
While I've never seen a codebase use the pre-operator newline style, e.g.

```ruby
one
  + two
  + three
```

...I suppose if that's preferred by your team, then there's no reason not to allow it. So allowing it on either side seems reasonable to me.

**Comment by srawlins** [comment]
I agree, it's kind of a weird style. I think people might split it that way if they're coming from [Google's Java Style Guide](https://google.github.io/styleguide/javaguide.html#s4.5.1-line-wrapping-where-to-break), which suggests breaking on non-assignment symbols before the symbol.
